src/passes/OptimizeInstructions.cpp

```diff
@@ -1199,6 +1259,9 @@ struct OptimizeInstructions
         auto type = curr->type;
         auto* left = curr->left->dynCast<Const>();
         auto* right = curr->right->dynCast<Const>();
+        // Note that we don't need to handle subtraction:
+        // canonicalization avoids a zero on the right anyhow, and a
+        // zero on the left can't be removed (it is how we negate ints).
```

Comment:
reading this now, I'm confused. why does canonicalization avoid a zero on the right? let's improve the comment, because if I can't figure it out now after just thinking about this a day or two ago, it will be confusing for other people I think...

I think a better comment could be `Canonicalization prefers an add instead of a subtract wherever possible. That prevents a subtracted constant on the right, as it would be added. And for a zero on the left, it can't be removed (it is how we negate ints).`